nixos/modules/services/misc/ethminer.nix

```diff
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ethminer;
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.ethminer = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "enable ethminer ether mining.";
+      };
+
+      recheck = mkOption {
+        type = types.int;
+        default = 2000;
+        description = "time in milliseconds to farm recheck.";
+      };
+
+      toolkit = mkOption {
+        type = types.enum [ "cuda" "opencl" ];
+        default = "cuda";
+        description = "cuda or opencl toolkit.";
+      };
+
+      apiPort = mkOption {
+        type = types.int;
+        default = -3333;
+        description = "ethminer api port. minus sign puts api in read-only mode.";
+      };
+
+      wallet = mkOption {
+        type = types.string;
+        example = "0x0484A1EfB3C88B0C7dE21919D480fF51099c8113";
```

Comment:
Where did you get this example from?

Reply:
https://etherscan.io/address/0x0484A1EfB3C88B0C7dE21919D480fF51099c8113 this is just an example of a real, but unused address.

Reply:
Eh.. I think it's best to just pick a random number here instead of an actual address